tests/integration/test_charm.py

```diff
@@ -11,90 +12,63 @@
 
 
-@pytest.mark.abort_on_fail
-async def test_build_and_deploy_with_alternative_images(ops_test, prometheus_charm):
-    """Test that the Prometheus charm can be deployed successfully."""
-    resources = {"prometheus-image": oci_image("./metadata.yaml", "prometheus-image")}
-    app_name = "prometheus-ubuntu"
-
-    await ops_test.model.deploy(prometheus_charm, resources=resources, application_name=app_name)
-    await ops_test.model.wait_for_idle(apps=[app_name], status="active")
-    await ops_test.model.block_until(lambda: len(ops_test.model.applications[app_name].units) > 0)
-
-    assert ops_test.model.applications[app_name].units[0].workload_status == "active"
-
-    await check_prometheus_is_ready(ops_test, app_name, 0)
-
-    await ops_test.model.applications[app_name].remove()
-    await ops_test.model.reset()
-
-
-@pytest.mark.abort_on_fail
-async def test_build_and_deploy_prometheus_tester(ops_test, prometheus_tester_charm):
-    """Test that Prometheus tester charm can be deployed successfully."""
-    resources = {
-        "prometheus-tester-image": oci_image(
-            "./tests/integration/prometheus-tester/metadata.yaml", "prometheus-tester-image"
-        )
-    }
-    app_name = "prometheus-tester"
-
-    await ops_test.model.deploy(
-        prometheus_tester_charm, resources=resources, application_name=app_name
-    )
-    await ops_test.model.wait_for_idle(apps=[app_name], status="active")
-    await ops_test.model.block_until(lambda: len(ops_test.model.applications[app_name].units) > 0)
-
-    assert ops_test.model.applications[app_name].units[0].workload_status == "active"
-
-    await ops_test.model.applications[app_name].remove()
-    await ops_test.model.reset()
-
-
 @pytest.mark.abort_on_fail
 async def test_prometheus_scrape_relation_with_prometheus_tester(
     ops_test, prometheus_charm, prometheus_tester_charm
 ):
     """Test basic functionality of prometheus_scrape relation interface."""
-    tester_resources = {
-        "prometheus-tester-image": oci_image(
-            "./tests/integration/prometheus-tester/metadata.yaml", "prometheus-tester-image"
-        )
-    }
-    prometheus_resources = {"prometheus-image": oci_image("./metadata.yaml", "prometheus-image")}
     prometheus_app_name = "prometheus"
     tester_app_name = "prometheus-tester"
-
-    await ops_test.model.deploy(
-        prometheus_charm, resources=prometheus_resources, application_name=prometheus_app_name
-    )
-    await ops_test.model.deploy(
-        prometheus_tester_charm, resources=tester_resources, application_name=tester_app_name
-    )
-    await ops_test.model.wait_for_idle(apps=[prometheus_app_name], status="active")
-    await ops_test.model.wait_for_idle(apps=[tester_app_name], status="active")
-    await ops_test.model.block_until(
-        lambda: len(ops_test.model.applications[prometheus_app_name].units) > 0
+    app_names = [prometheus_app_name, tester_app_name]
+
+    await asyncio.gather(
```

Comment:
Given the limited resources of ghwf, maybe we should avoid using gather for deployments.

Reply:
In my experience it makes no difference. After all, it's concurrent rather than parallel execution.